Refactor Stroke API and add MediaContainerNode.replaceMedia() API

xerxovksi opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

### Changes
- Refactor `Stroke` API
- Add `MediaContainerNode.replaceMedia()`

### Additional changes
- The [TypeDoc](https://www.npmjs.com/package/typedoc) package is used to generate the docs.
- It has been upgraded from version `0.23.15` to `0.25.13` for compatibility with TypeScript `5.4.5`.
- Given this change, the structure of the docs are slightly modified.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

hollyschinsky reviewed (changes requested):
@xerxovksi I noticed the names of the signatures are all shifted over to the right compared to our prod docs. 

<img width="1014" alt="Screen Shot 2024-09-09 at 6 00 55 PM" src="https://github.com/user-attachments/assets/9c0c218c-8fd9-46c6-ad1a-6d294c9eaef6">
vs
on prod
<img width="1023" alt="Screen Shot 2024-09-09 at 6 01 42 PM" src="https://github.com/user-attachments/assets/e0670f50-4583-4e23-a1e8-e5bff89e94c1">

Looks like maybe the little char that was there is gone too, but moreso the extra space should at least get removed I guess. Must be something in how they got converted since I see it in all of the doc API refs :/

xerxovksi commented:
> @xerxovksi I noticed the names of the signatures are all shifted over to the right compared to our prod docs.
> 
> <img alt="ss13" width="1014" src="https://private-user-images.githubusercontent.com/359935/365824759-8e664145-22f9-4216-96f2-f6ce715de403.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NDM0NjQsIm5iZiI6MTcyNTk0MzE2NCwicGF0aCI6Ii8zNTk5MzUvMzY1ODI0NzU5LThlNjY0MTQ1LTIyZjktNDIxNi05NmYyLWY2Y2U3MTVkZTQwMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMFQwNDM5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMGY4MzczZDZlYzA1MGRkNDU4YjQ0ZmUyOGRmOTE0YjZmOTZhZmQ5ODQ1NGY0OWM4ZTYxNWEzZWVjNjdjNzcwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ZtYb7M5XcMrrZl5DymHwYVsoQfcjeoNszh7en_DaSNY">
> vs on prod
> 
> <img alt="ss13" width="1014" src="https://private-user-images.githubusercontent.com/359935/365824790-31876fa8-d3bc-439d-8137-0271e7ca7c72.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NDM0NjQsIm5iZiI6MTcyNTk0MzE2NCwicGF0aCI6Ii8zNTk5MzUvMzY1ODI0NzkwLTMxODc2ZmE4LWQzYmMtNDM5ZC04MTM3LTAyNzFlN2NhN2M3Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMFQwNDM5MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MDMwNmU1NGI3MWZhNGQ0YWUyZTE2ZDk1MDViZGYxMWIyMjEwMDIzZWNiYzFhZjk0ZWMzZTZhYjc4MTNlMzZkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eqkGwqJf3zgggDnWBVeZxu1YMBvVgSML-c_4Z8wnvio">
> Looks like maybe the little char that was there is gone too, but moreso the extra space should at least get removed I guess. Must be something in how they got converted since I see it in all of the doc API refs :/

Yes, this is generated. I have removed the indentation now and added it to the manual steps in the README.

hollyschinsky commented:
Added entry to changelog and merging.
